From e6bfce202e77f8ac76c84721989148606e4b7105 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 15:21:50 -0400
Subject: [infiniband] ocrdma: Handle CQ overrun error

Message-id: <5ac1ce21913bdb766c2da9f68552b8a479f28c31.1411571481.git.dledford@redhat.com>
Patchwork-id: 96633
O-Subject: [Patch RHEL7 15/48] RDMA/ocrdma: Handle CQ overrun error
Bugzilla: 1081067 1088577
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Selvin Xavier <selvin.xavier@emulex.com>

Signed-off-by: Selvin Xavier <selvin.xavier@emulex.com>
Signed-off-by: Devesh Sharma <devesh.sharma@emulex.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 1228056bcff2e4029c3d9d5d31cf122eea3a4c6a)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_hw.c b/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
index f1ec9d9..ebcb260 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
@@ -640,7 +640,7 @@ static void ocrdma_dispatch_ibevent(struct ocrdma_dev *dev,
 {
  struct ocrdma_qp *qp = NULL;
  struct ocrdma_cq *cq = NULL;
- struct ib_event ib_evt;
+ struct ib_event ib_evt = { 0 };
  int cq_event = 0;
  int qp_event = 1;
  int srq_event = 0;
@@ -665,6 +665,8 @@ static void ocrdma_dispatch_ibevent(struct ocrdma_dev *dev,
  case OCRDMA_CQ_OVERRUN_ERROR:
   ib_evt.element.cq = &cq->ibcq;
   ib_evt.event = IB_EVENT_CQ_ERR;
+  cq_event = 1;
+  qp_event = 0;
   break;
  case OCRDMA_CQ_QPCAT_ERROR:
   ib_evt.element.qp = &qp->ibqp;
@@ -726,6 +728,7 @@ static void ocrdma_dispatch_ibevent(struct ocrdma_dev *dev,
            qp->srq->ibsrq.
            srq_context);
  } else if (dev_event) {
+  pr_err("%s: Fatal event received\n", dev->ibdev.name);
   ib_dispatch_event(&ib_evt);
  }
 
-- 
1.7.1